src/coreComponents/dataRepository/HistoryDataSpec.hpp

```diff
@@ -213,8 +213,13 @@ inline
 typename std::enable_if< ( traits::is_array_type< ARRAY_T >) && (ARRAY_T::NDIM > 1) && can_history_io< typename ARRAY_T::value_type >, HistoryMetadata >::type
 getHistoryMetadata( string const & name, ARRAY_T const & arr, localIndex sizeOverride = -1 )
 {
-  localIndex perIndexSize = arr[ 0 ].size( );
-  localIndex numIndices = ( sizeOverride >= 0 ? sizeOverride :  arr.size( ) / perIndexSize );
+  // Array dim > 1 so this should be valid
+  localIndex perIndexSize = arr.size( 1 );
+  localIndex numIndices = 0;
+  if( !arr.empty( ) )
+  {
+    numIndices = ( sizeOverride >= 0 ? sizeOverride :  arr.size( ) / perIndexSize );
+  }
```

Comment:
```suggestion
  localIndex const perIndexSize = arr.size( 1 );
  localIndex const numIndices = sizeOverride >= 0 ? sizeOverride :  arr.size( ) / perIndexSize ) ;
```
if `arr.empty()` then `arr.size()` is zero...which is the desired value?